src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -328,14 +318,14 @@ def request(self, url, params, headers=None, verb='GET',
         if verbose:
             print(verb, url, params, headers)
         header = self.parse_header(hbuf.getvalue())
+        data = bbuf.getvalue()
```

Comment:
This part should be checked if it works with `HEAD` HTTP body and different HTTP codes since you moved reading buffer before you check HTTP verb as it was initially implemented in line block below (lines 323-329)